lib/Graph/Grad.cpp

```diff
@@ -144,6 +144,33 @@ Function *glow::differentiate(Function *F, const TrainingConfig &conf,
       continue;
     }
 
+    if (N->getKind() == Kind::TileNodeKind) {
+      TileNode *TN = cast<TileNode>(N);
+      NodeValue outputG = map.getGradient(TN->getResult());
+
+      // To compute the gradient with respect to the input of the TileNode, all
+      // of the slices in outputG corresponding to the tiled slices in the
+      // foward pass need to be added together. This is achieved by reshaping
+      // outputG to replace the tiling axis with {numTiles, tileDim}, and then
```

Comment:
I feel like this is a confusing description but I don't know how to say it better. Maybe an example like if the tile created {n,x,h,w} from {n,c,h,w} where x = c * num_tiles then we reshape {n,x,h,w} to {n, num_tiles, c, h, w} so that BatchedReduceAddNode eliminates the num_tiles getting back to {n,c,h,w}